packages/playwright-test/src/matchers/expect.ts

```diff
@@ -236,22 +241,32 @@ class ExpectMetaInfoProxyHandler {
         }
 
         const serializerError = serializeError(jestError);
-        step.complete({ error: serializerError });
+        if (traceEvent) {
+          traceEvent.metadata.error = { error: { name: jestError.name, message: jestError.message, stack: jestError.stack } };
+          traceEvent.metadata.endTime = monotonicTime();
+          step.complete({ error: serializerError });
+        }
         if (this._info.isSoft)
           testInfo._failWithError(serializerError, false /* isHardError */);
         else
           throw jestError;
       };
 
+      const finalizer = () => {
+        if (traceEvent)
+          traceEvent.metadata.endTime = monotonicTime();
```

Comment:
We should probably update trace event title from `step.refinedTitle`.

Reply:
I match by apiName + wallTime now, so can't do.